src/libraries/System.Private.CoreLib/src/System/Text/Unicode/Utf16Utility.Validation.cs

```diff
@@ -87,6 +87,11 @@ static Utf16Utility()
                     Vector128<ushort> vectorA800 = Vector128.Create((ushort)0xA800);
                     Vector128<short> vector8800 = Vector128.Create(unchecked((short)0x8800));
                     Vector128<ushort> vectorZero = Vector128<ushort>.Zero;
+
+                    Vector128<byte> bitMask128 = BitConverter.IsLittleEndian ?
+                        Vector128.Create(0x80402010_08040201).AsByte() :
+                        Vector128.Create(0x01020408_10204080).AsByte();
```

Comment:
Does the JIT know that `Vector128.Create` is side-effect free so it can eliminate this code when the `bitMask128` variable is unused?  If not, this is adding unnecessary overhead to the x86 path.

Reply:
I'm not sure. @echesakovMSFT @tannergooding @kunalspathak do you know the answer to this?

Regardless of the answer, since the code was moved from being a static field (which was initialized for all architectures) to a local variable (which is also available for all architectures), I assume the behavior would not be worse than before.

If we think this needs to be addressed to avoid the unnecessary overhead in x86, I can create an issue so it gets fixed later.

I would like to get this merged as soon as possible to unblock net472 (prevent the PNS exception on runtime), so the backport to Prev8 gets approved.

Reply:
The static would only have been initialized once.  The local is created for every invocation.

The simple workaround, if indeed the JIT isn't eliminating it as dead, would just be to re-use the zero vector above so this one becomes an unused alias.  i.e.:

```C#
Vector128<byte> bitMask128 = !AdvSimd.IsSupported ? vectorZero :
    BitConverter.IsLittleEndian ?
    Vector128.Create(0x80402010_08040201).AsByte() :
    Vector128.Create(0x01020408_10204080).AsByte();
```

Reply:
Ah, thanks for the clarification, @saucecontrol . I'll add it to the list of TODOs to address in the next PR.

Reply:
We do eliminate dead stores. Here is a sample test on x64:
```csharp
Vector128<byte> bitMask128 = BitConverter.IsLittleEndian ?
            Vector128.Create(0x80402010_08040201).AsByte() :
            Vector128.Create(0x01020408_10204080).AsByte();

if (AdvSimd.IsSupported)
{
    Console.WriteLine(bitMask128);
}
else
{
    Console.WriteLine(1);
}
```
Produces:

```asm
G_M58174_IG01:
       4883EC28             sub      rsp, 40
                                                ;; bbWeight=1    PerfScore 0.25
G_M58174_IG02:
       B901000000           mov      ecx, 1
       E832FDFFFF           call     System.Console:WriteLine(int)
       B864000000           mov      eax, 100
                                                ;; bbWeight=1    PerfScore 1.50
G_M58174_IG03:
       4883C428             add      rsp, 40
       C3                   ret
                                                ;; bbWeight=1    PerfScore 1.25
```